src/Infrastructure/Service/Precondition/AbstractFileIteratingPrecondition.php

```diff
@@ -57,11 +57,7 @@ public function isFulfilled(
                 foreach ($files as $file) {
                     $file = $this->pathFactory::create($file);
 
-                    if (!$this->filesystem->isLink($file)) {
-                        continue;
-                    }
```

Comment:
Removing this check pushes the responsibility of limiting scope down to implementations. It also reduces the number of filesystem calls.